redis/src/commands/json.rs

```diff
@@ -0,0 +1,231 @@
+	fn json_arr_index<K: ToRedisArgs, P: ToRedisArgs, V: Serialize>(key: K, path: P, value: &'a V) {
+		match serde_json::to_string(value) {
+			Err(e) => {
+				json_serialize_fail!(e)
+			},
+			Ok(serialised) => {
+				Ok::<Cmd, RedisError>(
+					cmd("JSON.ARRINDEX")
+						.arg_take(key)
+						.arg_take(path)
+						.arg_take(serialised)
+				)
+			}
+		}
+	}
+
+	/// Inserts the JSON `value` in the array at `path` before the `index` (shifts to the right)
+	/// `index` must be withing the array's range.
+	fn json_arr_insert<K: ToRedisArgs, P: ToRedisArgs, V: Serialize>(key: K, path: P, index: i64, value: &'a V) {
+		match serde_json::to_string(value) {
+			Err(e) => {
+				json_serialize_fail!(e)
+			},
+			Ok(serialised) => {
+				Ok::<Cmd, RedisError>(
+					cmd("JSON.ARRINSERT")
+						.arg_take(key)
+						.arg_take(path)
+						.arg_take(index)
+						.arg_take(serialised)
+				)
+			}
+		}
+	}
+
+	/// Reports the length of the JSON Array at `path` in `key`.
+	fn json_arr_len<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P) {
+		Ok::<Cmd, RedisError>(
+			cmd("JSON.ARRLEN")
+				.arg_take(key)
+				.arg_take(path)
+		)
+	}
+
+	/// Removes and returns an element from the `index` in the array.
+	/// `index` defaults to `-1` (the end of the array)
+	fn json_arr_pop<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P, index: i64) {
+		Ok::<Cmd, RedisError>(
+			cmd("JSON.ARRPOP")
+				.arg_take(key)
+				.arg_take(path)
+				.arg_take(index)
+		)
+	}
+
+	/// Trims an array so that it contains only the specified inclusive range of elements.
+	///
+	/// This command is extremely forgiving and using it with out-of-range indexes will not produce an error.
+	/// There are a few differences between how RedisJSON v2.0 and legacy versions handle out-of-range indexes.
+	fn json_arr_trim<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P, start: i64, stop: i64) {
+		Ok::<Cmd, RedisError>(
+			cmd("JSON.ARRTRIM")
+				.arg_take(key)
+				.arg_take(path)
+				.arg_take(start)
+				.arg_take(stop)
+		)
+	}
+
+	/// Clears container values (Arrays/Objects), and sets numeric values to 0.
+	fn json_clear<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P) {
+		Ok::<Cmd, RedisError>(
+			cmd("JSON.CLEAR")
+				.arg_take(key)
+				.arg_take(path)
+		)
+	}
+
+	/// Deletes a value at `path`.
+	fn json_del<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P) {
+		Ok::<Cmd, RedisError>(
+			cmd("JSON.DEL")
+				.arg_take(key)
+				.arg_take(path)
+		)
+	}
+
+	/// Gets JSON Value(s) at `path`
+	/// Runs JSON.GET is key is singular, Runs JSON.MGET if there are multiple keys
+	fn json_get<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P) {
+		Ok::<Cmd, RedisError>(
+			cmd(if key.is_single_arg() { "JSON.GET" } else { "JSON.MGET" })
+				.arg_take(key)
+				.arg_take(path)
+		)
+	}
+
+	/// Increments the number value stored at `path` by `number`.
+	fn json_numincrby<K: ToRedisArgs, P: ToRedisArgs>(key: K, path: P, value: i64) {
```

Comment:
Nit: should be named `json_num_incr_by`. Same for all the other method names, so `objkeys` -> `obj_keys` and `strlen` -> `str_len`.